### PR TITLE
ADM-898[backend][frontend]: feat: block' in board mapping is mutually exclusive with 'flag as block'

### PR DESCRIPTION
**Pull request by yulongcai**
## Summary

block' in board mapping is mutually exclusive with 'flag as block'
## Before

block mapping and flag as block all check
image
image

## After

block mapping and flag as block only check one
image
image

## Note

_Null_

### CODE REVIEW

File: backend/src/test/java/heartbeat/util/BoardUtilTest.java [hunk 1]
```diff
@@ -81,16 +81,29 @@ void calculateCycleTimeWhenTreatFlagCardAsBlockIsFalse() {
...
+	@Test
+	void shouldCalculateOriginCycleTimeGivenTreatFlagCardAsBlockIsFalse() {
+		List<StatusChangedItem> statusChangedItems = StatusChangedItemsListAndCycleTimeInfosListFixture
+			.STATUS_CHANGED_ITEMS_LIST_OF_ORIGIN();
+		List<CycleTimeInfo> statusChangedItemsWithoutFlagExpect = StatusChangedItemsListAndCycleTimeInfosListFixture
+			.CYCLE_TIME_INFOS_LIST_OF_ORIGIN_WITHOUT_FLAG();
+
+		when(workDay.calculateWorkDaysBy24Hours(anyLong(), anyLong()))
+			.thenReturn(StatusChangedItemsListAndCycleTimeInfosListFixture.EXPECT_DAYS);
+		List<CycleTimeInfo> result = boardUtil.getOriginCycleTimeInfos(statusChangedItems, Boolean.FALSE);
```
Comment (guzhongren) [inline backend/src/test/java/heartbeat/util/BoardUtilTest.java:105]: new line between `given when then`

File: frontend/__tests__/containers/MetricsStep/CycleTime.test.tsx [hunk 2]
```diff
@@ -406,4 +406,38 @@ describe('CycleTime', () => {
       expect(mockedUseAppDispatch).not.toHaveBeenCalledWith(saveDoneColumn([]));
     });
   });
+
+  it('should show warning message when both mapping block column and add flag as block Given mapping relationship by status', () => {
```
Comment (guzhongren) [inline frontend/__tests__/containers/MetricsStep/CycleTime.test.tsx:410]: when -> given,

File: frontend/__tests__/containers/MetricsStep/CycleTime.test.tsx [hunk 3]
```diff
@@ -406,4 +406,38 @@ describe('CycleTime', () => {
...
+          value: 'Block',
+        },
+      ],
+    });
+    setup();
+
+    expect(screen.getByText(MESSAGE.FLAG_CARD_DROPPED_WARNING)).toBeVisible();
+  });
+
+  it('should show warning message when both mapping block column and add flag as block Given mapping relationship by column', () => {
```
Comment (guzhongren) [inline frontend/__tests__/containers/MetricsStep/CycleTime.test.tsx:427]: when

File: frontend/src/utils/util.ts [hunk 4]
```diff
@@ -156,3 +156,12 @@ export function convertCycleTimeSettings(
   }
   return cycleTimeSettings?.map(({ status, value }: ICycleTimeSetting) => ({ [status]: value }));
 }
+
+export function existBlockColumn(
+  cycleTimeSettingsType: CYCLE_TIME_SETTINGS_TYPES,
+  cycleTimeSettings: ICycleTimeSetting[],
+) {
+  return cycleTimeSettingsType === CYCLE_TIME_SETTINGS_TYPES.BY_COLUMN
+    ? cycleTimeSettings.some(({ column }) => column.toUpperCase() === 'BLOCKED' || column.toUpperCase() === 'BLOCK')
```
Comment (guzhongren) [inline frontend/src/utils/util.ts:165]: 'BLOCKED' and others should be a constant

File: backend/src/test/java/heartbeat/util/StatusChangedItemsListAndCycleTimeInfosListFixture.java [hunk 5]
```diff
@@ -18,8 +18,8 @@ public static List<StatusChangedItem> STATUS_CHANGED_ITEMS_LIST_OF_REAL_DONE_COL
 
 	public static List<CycleTimeInfo> CYCLE_TIME_INFOS_LIST_OF_REAL_DONE_COLUMN() {
 		return List.of(CycleTimeInfo.builder().column("DONE").day(EXPECT_DAYS).build(),
-				CycleTimeInfo.builder().column("BLOCK").day(0.0).build(),
-				CycleTimeInfo.builder().column("IN PROGRESS").day(EXPECT_DAYS).build());
+				CycleTimeInfo.builder().column("IN PROGRESS").day(EXPECT_DAYS).build(),
```
Comment (guzhongren) [inline backend/src/test/java/heartbeat/util/StatusChangedItemsListAndCycleTimeInfosListFixture.java:21]: `IN PROGRESS` and others should be a constant...

File: backend/src/test/java/heartbeat/util/BoardUtilTest.java (hunk 1 above)
Comment (yulongcai) [reply backend/src/test/java/heartbeat/util/BoardUtilTest.java:106]: done

File: backend/src/test/java/heartbeat/util/StatusChangedItemsListAndCycleTimeInfosListFixture.java (hunk 5 above)
Comment (yulongcai) [reply backend/src/test/java/heartbeat/util/StatusChangedItemsListAndCycleTimeInfosListFixture.java]: done

File: frontend/__tests__/containers/MetricsStep/CycleTime.test.tsx (hunk 3 above)
Comment (TingyuDong) [reply frontend/__tests__/containers/MetricsStep/CycleTime.test.tsx:427]: Solved.

File: frontend/__tests__/containers/MetricsStep/CycleTime.test.tsx (hunk 2 above)
Comment (TingyuDong) [reply frontend/__tests__/containers/MetricsStep/CycleTime.test.tsx:410]: Solved.

File: frontend/src/utils/util.ts (hunk 4 above)
Comment (TingyuDong) [reply frontend/src/utils/util.ts]: Solved.

**Review by neomgb — APPROVED** [review]

**Pull request merged**